google-cloud-storage/acceptance/storage/bucket_policy_test.rb

```diff
@@ -70,13 +70,54 @@
       policy.roles.count.must_equal 2
       expect { policy.bindings }.must_raise RuntimeError
 
+      policy = bucket_policy_v3.policy requested_policy_version: 1
+      policy.must_be_kind_of Google::Cloud::Storage::Policy
+      policy.version.must_equal 1
+      expect { policy.roles }.must_raise RuntimeError
```

Comment:
If you request version 1 shouldn't you be using `roles` instead of `bindings`?

Reply:
I'm pretty sure this is correct as written. 

@jkwlui or @frankyn: Can you confirm?

Reply:
Yeah this doesn't make a lot of sense to me. If you request version 1 explicitly, it doesn't make sense that you'd get a V3 object. You should get a V1 object (or an error that the underlying data is V3 on the request).

Reply:
Apologies for the delay. This is the expectation based on our conversation last week.

The rationale for doing this was, if a user is explicitly opting in with requested_policy_only with 1 or 3 then the client should respond with PolicyV3 class with the Array datastructure rather than the existing implementation.

Reply:
After offline discussion with @jkwlui, @frankyn, @blowmage and @dazuma, changing to return `PolicyV1` object when `requested_policy_version: 1`.